core/src/test/java/org/elasticsearch/cluster/metadata/IndexGraveyardTests.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.ByteBufferStreamInput;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.lessThan;
+
+/**
+ * Tests for the {@link IndexGraveyard} class
+ */
+public class IndexGraveyardTests extends ESTestCase {
+
+    public void testIsEquals() {
+        final IndexGraveyard graveyard1 = new IndexGraveyard();
+        assertTrue("empty graveyards are equal", graveyard1.isEquals(new IndexGraveyard(graveyard1)));
+        final IndexGraveyard graveyard2 = createRandom();
+        assertTrue("same graveyards are equal", graveyard2.isEquals(new IndexGraveyard(graveyard2)));
+    }
+
+    public void testSerialization() throws IOException {
+        final int numIterations = 5;
+        for (int i = 0; i < numIterations; i++) {
```

Comment:
We don't need the poor man's iterations here, this will run frequently enough on CI. The problem with loops like this in tests is that reproducibility can be broken if there is any non-determinism within the loop that causes random draws to occur a non-deterministic number of times. I'm not saying one way or the other whether or not that is the case here, but it has been a problem before, and it's easier to just avoid the problem altogether if the additional iterations are not buying us anything.